test/intl402/DateTimeFormat/prototype/formatRangeToParts/pattern-on-calendar.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google Inc.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-initializedatetimeformat
+description: Checks the DateTimeFormat choose different patterns based
+    on calendar.
+locale: [en]
+---*/
+
+let calendars = [
+  "buddhist",
+  "chinese",
+  "coptic",
+  "dangi",
+  "ethiopic",
+  "ethioaa",
+  "gregory",
+  "hebrew",
+  "indian",
+  "islamic",
+  "islamic-civil",
+  "islamic-rgsa",
+  "islamic-tbla",
+  "islamic-umalqura",
+  "japanese",
+  "persian",
+  "roc"
+];
+let date1 = new Date(2017, 3, 12);
+let date2 = new Date();
+
+// serialize parts to a string by considering only the type and literal.
+function serializeTypesAndLiteral(parts) {
+  var types = new Array();
+  parts.map(part => {
+    if (part.type == "literal") {
+      types.push(part.type + "(" + part.value + ")");
+    } else {
+      types.push(part.type);
+    }
+  });
```

Comment:
Sorry I wasn't clearer before; I meant something like

```js
let types = parts.map(part => {
  if (part.type == "literal") {
    return part.type + "(" + part.value + ")";
  }

  return part.type;
});
```

Reply:
Why not use template string?

Reply:
That would work as well.